slang.h

```diff
@@ -810,6 +829,51 @@ extern "C"
 
     #define SLANG_UUID_ISlangFileSystemExt { 0x5fb632d2, 0x979d, 0x4481, { 0x9f, 0xee, 0x66, 0x3c, 0x3f, 0x14, 0x49, 0xe1 } }
 
+    /* Identifies different types of writer target*/
+    typedef unsigned int SlangWriterChannel;
+    enum
+    {
+        SLANG_WRITER_CHANNEL_DIAGNOSTIC,
+        SLANG_WRITER_CHANNEL_STD_OUTPUT,
+        SLANG_WRITER_CHANNEL_STD_ERROR,
+        SLANG_WRITER_CHANNEL_COUNT_OF,
```

Comment:
Yeah, I'm not arguing against the value of the idiom. 99% of my objection is that I don't like `COUNT_OF` or `CountOf` as the name for this because of how far it stretches grammatical convention. In the case of C-style `enum`s with my usual convention, I find that just `Count` can be made very clear:

```c++
enum WriterChannel {
    kWriterChannel_Diagnostic,
    kWriterChannel_StandardOutput,
    kWriterChannelCount,
};
```

By eliminating the `_`, the `Count` case both reads naturally, and is easily distinguished between the ordinary cases so that you are less likely to get confused about its role.

In an `enum class` or a `SCREAMING_SNAKE_CASE` enumeration there is less ability to distinguish the "count" tag as distinct from the others, so I get the reasoning for making it stand out like a sore thumb with something like `CountOf` or `COUNT_OF`, but it also grates when reading it because "the X count` scans, as does "the count of X's," but "the X count of" just leaves me thinking "of *what*?" every time.

I would strongly prefer to just stick with `COUNT` over `COUNT_OF` while recognizing that it makes slip-ups slightly more likely. If there is a better alternative I'm open to it.

There's a smaller concern over source compatibility, in that a user might have written code that works with one value for `SLANG_WRITER_COUNT_OF` and that either doesn't compile, or compiles but is semantically correct, when the header changes to expose a new value. That is a weaker concern because a user *can* code defensively around such things, and in particular having their code break when the API adds a new case for them to deal with sounds like a feature rather than a bug.